.github/workflows/soak-testing.yml

```diff
@@ -0,0 +1,233 @@
+  # Add this?
+  # RATE: TPS
+
+jobs:
+  test_apps_and_publish_results:
+    name: Publish app and Soak Test - (${{ matrix.app-platform }}, ${{ matrix.instrumentation-type }})
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      packages: write
+      issues: write
+    strategy:
+      fail-fast: false
+      matrix:
+        app-platform: [ flask ]
+        # instrumentation-type: [ auto, manual ]
+        instrumentation-type: [ auto ]
+    env:
+      IMAGE_SUFFIX: sample-app-${{ matrix.app-platform }}-${{ matrix.instrumentation-type }}-${{ github.sha }}
+    steps:
+      - name: Configure Soak Test environment variables
+        # TODO: Does the LOGS_NAMESPACE need a `-$GITHUB_RUN_ID`?
+        run: >-
+          echo "LOGS_NAMESPACE=${{ github.repository }}/soak-tests" >> $GITHUB_ENV;
+          echo "NUM_OF_CPUS=$(nproc --all)" >> $GITHUB_ENV;
+      - uses: actions/checkout@v2
+      - name: Log in to the GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+      - name: Extract metadata (tags, labels) for Docker
+        id: docker-image-metadata
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ghcr.io/${{ github.repository }}-${{ env.IMAGE_SUFFIX }}
+      # TODO: Could we just build in `docker-compose` build/context without publishing?
+      - name: Build and Push Docker image
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          context: integration-test-apps/${{ matrix.instrumentation-type}}-instrumentation/${{ matrix.app-platform }}
+          tags: ${{ steps.docker-image-metadata.outputs.tags }}
+          labels: ${{ steps.docker-image-metadata.outputs.labels }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.AWS_REGION }}
+      - name: Prepare Metric Data Queries JSON Inputs
+        run: >-
+          for metric_kind in cpu-load total-memory;
+          do
+            echo "$(
+              cat ./.github/aws-cloudwatch-api-values/get-metric-data/$metric_kind-metric-data-query.json |
+              sed -e "s/<METRIC_PERIOD>/${{ env.HOSTMETRICS_COLLECTION_INTERVAL_IN_SECONDS }}/g" \
```

Comment:
I think you can prepare the JSON in python to make it more simple than sedding, and just accept these variables as program args.

Reply:
Sure I can make more Python Scripts. I liked using `sed` because it put everything upfront in one place (the `soak-testing.yml` file) but the Python script is arguably cleaner and therefore easier to understand than `sed`.